subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/SnapshotUtil.java

```diff
@@ -193,9 +250,21 @@ private static boolean isRegularFileOrDirectory(MetadataSnapshot metadataSnapsho
     }
 
     private interface DescendantHandler<T> {
+        /**
+         * relativePath is a descendant of prefix.
+         */
         T handleDescendant();
+        /**
+         * relativePath is a parent of prefix.
+         */
         T handleParent();
+        /**
+         * relativePath is the same as prefix.
+         */
         T handleSame();
+        /**
+         * relativePath has a common prefix with prefix, but is different.
```

Comment:
Can `commonPrefixLenght` be `0`?

Reply:
Yeah, the `commonPrefixLength` can be `0`.

Reply:
I think that would be worth clarifying. I'd also argue that if `commonPrefixLength == 0` then the paths do _not_ have a common prefix, which contradicts the Javadoc.

Reply:
I changed the Javadoc.